lib/client/api_test.go

```diff
@@ -709,3 +710,51 @@ func TestVirtualPathNames(t *testing.T) {
 		})
 	}
 }
+
+func Test_formatConnectToProxyErr(t *testing.T) {
+	tests := []struct {
+		name string
+		err  error
+
+		wantError       string
+		wantUserMessage string
+	}{
+		{
+			name: "nil error passes through",
+			err:  nil,
+		},
+		{
+			name:      "unrelated error passes through",
+			err:       fmt.Errorf("flux capacitor undercharged"),
+			wantError: "flux capacitor undercharged",
+		},
+		{
+			name:            "principals mismatch user message injected",
+			err:             fmt.Errorf(`ssh: handshake failed: ssh: principal "" not in the set of valid principals for given certificate`),
+			wantError:       `ssh: handshake failed: ssh: principal "" not in the set of valid principals for given certificate`,
+			wantUserMessage: unconfiguredPublicAddrMsg,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := formatConnectToProxyErr(tt.err)
+			if tt.wantError == "" {
+				require.NoError(t, err)
+				return
+			}
+			traceErr, isTraceErr := err.(*trace.TraceErr)
+
+			if !isTraceErr {
+				require.EqualError(t, err, tt.wantError)
+			} else {
+				require.EqualError(t, traceErr.OrigError(), tt.wantError)
+			}
```

Comment:
nit: test for the positive, since we have both arms in the if:

```suggestion
			if isTraceErr {
				require.EqualError(t, traceErr.OrigError(), tt.wantError)
			} else {
				require.EqualError(t, err, tt.wantError)
			}
```